Library/Homebrew/rubocops/urls.rb

```diff
@@ -217,6 +217,13 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
             problem "Use /archive/ URLs for GitHub tarballs (url is #{url})."
           end
 
+          archive_ref_tags_gh_pattern = %r{https://.*github.*/archive/(?![a-fA-F0-9]{40})(?!refs/tags/).*\.tar\.gz$}
+          audit_urls(urls, archive_ref_tags_gh_pattern) do |_, url|
+            next if url.end_with?(".git")
+
+            problem "Use /archive/refs/tags URLs for GitHub tarballs (url is #{url})."
```

Comment:
Can these both trigger at once?

Reply:
i assume you are talking about the `archive_gh_pattern`, i just tested, looks like it cannot be triggered all together

Reply:
Cool, thanks 👍🏻

Reply:
Perhaps also worth checking it won't suggest a change to a format rejected by this audit.

Reply:
oh, i see your point, that is right. let me add a fix

Reply:
actually could be a followup pr to autocorrect the fix.